test/src/main/java/org/axonframework/test/aggregate/ResultValidator.java

```diff
@@ -274,6 +275,100 @@
+     * duration}.
+     *
+     * @param duration the time at which no deadline matching the given {@code matcher} should be scheduled
+     * @param matcher  the matcher defining the deadline which should not be scheduled
+     * @return the current ResultValidator, for fluent interfacing
+     */
+    ResultValidator<T> expectNoScheduledDeadlineMatching(Duration duration,
+                                                         Matcher<? super DeadlineMessage<?>> matcher);
+
+    /**
+     * Asserts that <b>no</b> deadline equal to the given {@code deadline} has been scheduled after the given {@code
+     * duration}.
+     * <p/>
+     * Note that the source attribute of the deadline is ignored when comparing deadlines. Deadlines are compared using
+     * an "equals" check on all fields in the deadlines.
+     *
+     * @param duration the time at which no deadline equal to the given {@code deadline} should be scheduled
+     * @param deadline the deadline which should not be scheduled
+     * @return the current ResultValidator, for fluent interfacing
+     */
+    ResultValidator<T> expectNoScheduledDeadline(Duration duration, Object deadline);
+
+    /**
+     * Asserts that <b>no</b> deadline of the given {@code deadlineType} has been scheduled after the given {@code
+     * duration}.
+     *
+     * @param duration     the time at which no deadline of {@code deadlineType} should be scheduled
+     * @param deadlineType the type of the deadline which should not be scheduled
+     * @return the current ResultValidator, for fluent interfacing
+     */
+    ResultValidator<T> expectNoScheduledDeadlineOfType(Duration duration, Class<?> deadlineType);
+
+    /**
+     * Asserts that <b>no</b> deadline with the given {@code deadlineName} has been scheduled after the given {@code
+     * duration}.
+     *
+     * @param duration     the time at which no deadline of {@code deadlineName} should be scheduled
+     * @param deadlineName the name of the deadline which should not be scheduled
+     * @return the current ResultValidator, for fluent interfacing
+     */
+    ResultValidator<T> expectNoScheduledDeadlineWithName(Duration duration, String deadlineName);
+
+    /**
+     * Asserts that <b>no</b> deadline matching the given {@code matcher} has been scheduled at the given {@code
+     * scheduledTime}.
+     * <p/>
+     * If the {@code scheduledTime} is calculated based on the "current time", use the {@link
+     * TestExecutor#currentTime()} to get the time to use as "current time".
+     *
+     * @param scheduledTime the time at which no deadline matching the given {@code matcher} should be scheduled
+     * @param matcher       the matcher defining the deadline which should not be scheduled
+     * @return the current ResultValidator, for fluent interfacing
+     */
+    ResultValidator<T> expectNoScheduledDeadlineMatching(Instant scheduledTime,
```

Comment:
The timestamp of a DeadlineMessage should reflect the time of expiry of the deadline. At least, the DeadlineManager implementations for production use, have the schedule date as the event timestamp once they publish the Deadline Message. It would be nice for the StubDeadlineManager to reflect this behavior.

That would make the Instant parameter obsolete for this method. As a result, the method could become more powerful, as one can more easily validate the absence of _any_ deadlines that match certain criteria.

Reply:
Very good point @abuijze! I've made the suggested change in [this](https://github.com/AxonFramework/AxonFramework/pull/1308/commits/560e6ab4882bbe3d16ee38a2287fa8daa59637f9) commit. Would you mind giving your opinion on it?